### PR TITLE
process several cases of string response for uart platrom

### PR DESCRIPTION
**Pull request by wesleywesley**
besides the success case string response, there are also several
other string patterns, we should process it instead of treat it as
bus error

when test with configure file of management ep (instead of nt ep),
which have no "Port Event Summary Register" in pff, therefore, the
read to this offset in gas have different response string.

### CODE REVIEW

File: lib/platform/linux-uart.c [hunk 1]
```diff
@@ -203,22 +264,45 @@ static void uart_gas_read(struct switchtec_dev *dev, void *dest,
...
+
+		/* case 2 */
+		if (strchr(gas_rd_rtn, ',')) {
+			if (sscanf(gas_rd_rtn,
+				 "%*[^<]<0x%x> [%d Byte]%*[^,],%*[^:]: 0x"
+				 "%x%*[^:]:",
+				  &raddr, &rnum, &rcrc) != 3)
+			continue;
+		}
+		else {
```
Comment (lsgunth) [inline lib/platform/linux-uart.c]: Nit: else belongs on the same line as the end brace

File: lib/platform/linux-uart.c [hunk 2]
```diff
@@ -273,6 +357,7 @@ static void uart_gas_write(struct switchtec_dev *dev, void __gas *dest,
 	addr = htobe32(addr);
 
 	for (i = 0; i< RETRY_NUM; i++) {
+
```
Comment (lsgunth) [inline lib/platform/linux-uart.c]: Nit: don't add this unnecessary line

File: lib/platform/linux-uart.c [hunk 3]
```diff
@@ -282,10 +367,22 @@ static void uart_gas_write(struct switchtec_dev *dev, void __gas *dest,
...
+		if (strstr(gas_wr_rtn, "Error with gas_reg_write()"))
+			break;
 
+		/* case 2 */
+		if (strchr(gas_wr_rtn, ',')) {
+			if (sscanf(gas_wr_rtn, "%*[^,],%*[^:]: [0x%x/0x%x]%*[^:]:",
+				   &cal, &exp) != 2)
+				continue;
+		}
+		else {
```
Comment (lsgunth) [inline lib/platform/linux-uart.c]: Nit: same as above

**Comment by wesleywesley** [comment]
@lsgunth 
thx for review. updated.

File: lib/platform/linux-uart.c [hunk 4]
```diff
@@ -271,8 +354,8 @@ static void uart_gas_write(struct switchtec_dev *dev, void __gas *dest,
 		crc = crc8((uint8_t *)src + i - 1, sizeof(uint8_t), crc, false);
 
 	addr = htobe32(addr);
-
 	for (i = 0; i< RETRY_NUM; i++) {
+
```
Comment (lsgunth) [inline lib/platform/linux-uart.c]: Still got a bunch of unrelated whitespace changes...

File: lib/platform/linux-uart.c [hunk 5]
```diff
@@ -203,22 +264,44 @@ static void uart_gas_read(struct switchtec_dev *dev, void *dest,
...
+		else
+			pos += 2;
+
 		for (j = 0; j < n; j++)
 			*ptr++ = strtol(pos, &pos, 16);
 
 		addr = htobe32(addr);
 		cal = crc8((uint8_t *)&addr, sizeof(addr), 0, true);
 		cal = crc8(dest, n, cal, false);
+
```
Comment (lsgunth) [inline lib/platform/linux-uart.c]: Ditto

**Comment by wesleywesley** [comment]
@lsgunth updated, pls approve.
